MetaMorpheus/GuiFunctions/Databases/DownloadUniProtDatabaseFunctions.cs

```diff
@@ -11,32 +12,19 @@ public class DownloadUniProtDatabaseFunctions
         public static string GetUniProtHtmlQueryString(string proteomeID, bool reviewed, bool isoforms, bool xmlFormat, bool compressed)
         {
             StringBuilder htmlQueryString = new StringBuilder();
-            htmlQueryString.Append("https://rest.uniprot.org/uniprotkb/stream?");
+            htmlQueryString.Append("https://rest.uniprot.org/uniprotkb/search?");
 
-            if (!xmlFormat) //fasta
-            {
-                htmlQueryString.Append("format=fasta");
-            }
-            else
-            {
-                htmlQueryString.Append("format=xml");
-            }
+            string[] queryArray = new string[4];
 
-            if (isoforms && !xmlFormat) //Only the .fasta file can be isoforms
-            {
-                htmlQueryString.Append("&includeIsoform=true");
-            }
+            queryArray[0] = compressed ? "compressed=true" : "";
+            queryArray[1] = xmlFormat ? "format=xml" : "format=fasta";
+            queryArray[2] = isoforms && !xmlFormat ? "includeIsoform=true" : "";
+            queryArray[3] = reviewed ? $"query=reviewed:true&proteome:{proteomeID}" : $"query=proteome:{proteomeID}";
 
-            if (reviewed)
-            {
-                htmlQueryString.Append("&reviewed=true");
-            }
+            string[] queryArrayReduced = queryArray.Where(x => x != "").ToArray() ;
```

Comment:
You don't need to convert back to an array here, you can leave it as an IEnumerable.

Also, you could set unneeded fields as null instead of "". I think a null check is slightly faster than string equality comparison, but it's a small difference.